streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -44,66 +49,120 @@ public static StoreFactory create(final String name) {
                 return new ValueFactory<K>() {
                     @Override
                     public <V> KeyValueFactory<K, V> withValues(final Serde<V> valueSerde) {
+
                         return new KeyValueFactory<K, V>() {
+                            public LoggedKeyValueFactory<K, V> logged(final String cleanupPolicy, final Map<String, String> config) {
+                                if (!InternalTopicManager.isValidCleanupPolicy(cleanupPolicy)) {
+                                    throw new IllegalArgumentException("cleanup policy: " + cleanupPolicy + " is not valid");
+                                }
+                                final Map<String, String> logConfig = new HashMap<>(config);
+                                logConfig.put("cleanup.policy", cleanupPolicy.toLowerCase(Locale.ROOT));
+                                return loggedKeyValueFactory(name, true, logConfig, keySerde, valueSerde);
+                            }
+
+                            public LoggedKeyValueFactory<K, V> notLogged() {
```

Comment:
I was actually thinking about adding the function in both `InMemory...` and `Persistent`. I admit it does introduce duplicated code, just thought it may look nicer on the user-facing APIs, as developers to specify their stores as:

```
Stores.create(...).withKey(...).withValue(...)
                          .persistent(...).windowed(...)
                          .enableCaching(...)
                          .enableLogging(...)
```

Thinking it a bit more, now I'm pondering if that's a good idea since by default we may want logging to be enabled (?maybe?), while with the `enableLogging` then by default logging is disabled. What do you think?